examples/agent-pool-terraform/userdata-ent

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# Source variables about OS version
+source /etc/os-release
+# Add the Teleport YUM repository. You'll need to update this file for each
+# major release of Teleport. First, get the major version from $VERSION_ID so
+# this fetches the correct package version.
+VERSION_ID=$(echo $VERSION_ID | grep -Eo "^[0-9]+")
+
+# Install the config-manager dnf plugin
+dnf install -y 'dnf-command(config-manager)'
+# Use the dnf config manager plugin to add the Teleport RPM repo
+dnf config-manager --add-repo "$(rpm --eval "https://yum.releases.teleport.dev/$ID/$VERSION_ID/Teleport/%_arch/stable/v${teleport_version}/teleport.repo")"
+
+# Install teleport
+dnf install -y teleport-ent
```

Comment:
So this will work only on RPM-based systems, which will be strange if an enterprise script would only support one type of systems while the OSS one had a proper install script.

Wouldn't a better way be updating our `https://goteleport.com/static/install.sh` install script to support passing an edition to it to take advantage of all its existing install logic?

Reply:
The Terraform module deploys an Amazon Linux 2023 instance, so these commands are meant to support that instance. The guide also [explains](https://github.com/gravitational/teleport/blob/2bab381f93ae453c9e98d08c9f121b60fde58a88/docs/pages/agents/deploy-agents-terraform.mdx#L151-L155) the need to edit the user data script for other environments.

I agree that it would be better to take advantage of the OS branching logic in `install.sh`, though. I'll look into getting the script edited.